### PR TITLE
YOMA-476 - refactor(api): remove old api module boilerplate

### PR DESCRIPTION
**Pull request by cameronolivier**
## Scope [YOMA-476](https://yomaworld.atlassian.net/browse/YOMA-476)
- remove old api module functions and boilerplate

## Work Done
- deleted all files in api module that we no longer need.
- cleaned up utils and code that's not required
- cleaned up code in the app where the api was being used.

## Steps to Test
- run tests.
- build

### CODE REVIEW

File: src/modules/Auth/Auth.middleware.test.ts [hunk 1]
```diff
@@ -203,73 +181,73 @@ describe('modules/Auth/Auth.middleware', () => {
...
-      const { store, invoke } = create(SUT.authRegistrationFlow(mockApi))
-
-      // when ... we respond to the failure state
-      await invoke(action)
-
-      // then ... the failure action should be called with the error response
-      expect(store.dispatch).toHaveBeenCalledWith(authRegistrationFailure(response))
-    })
-  })
+  // describe('authRegistrationFlow', () => {
```
Comment (cameronolivier) [inline src/modules/Auth/Auth.middleware.test.ts]: commenting this out - as it should be overwritten by the new registration work

File: src/modules/Auth/Auth.middleware.ts [hunk 2]
```diff
@@ -91,28 +90,16 @@ export const authLoginFailureFlow =
...
-export const authRegistrationFlow =
-  ({ api }: { api: any }): Middleware =>
-  ({ dispatch }) =>
-  next =>
-  async action => {
-    const result = next(action)
+export const authRegistrationFlow: Middleware = _store => next => async action => {
+  const result = next(action)
 
+  if (authRegistration.match(action)) {
```
Comment (cameronolivier) [inline src/modules/Auth/Auth.middleware.ts]: removed the rest of this -but it'll be fixed in the new PR with the registration work

**Review by sciphergfx — DISMISSED** [review]
Looks Good

**Review by sciphergfx — APPROVED** [review]
All good

**Pull request merged**